app/src/setupProxy.js

```diff
@@ -1,22 +1,40 @@
 const { createProxyMiddleware } = require("http-proxy-middleware");
+const express = require("express");
+const path = require("path");
 
-module.exports = function (app) {
-    app.use(
-        [
-            "/api",
-            "/admin-ng",
-            "/acl-manager",
-            "/info",
-            "/services",
-            "/sysinfo",
-            "/staticfiles",
-            "/j_spring_security_check",
-            "/ui",
-        ],
-        createProxyMiddleware({
-            target: process.env.PROXY_TARGET || "http://localhost:5000",
-            changeOrigin: true,
-            auth: process.env.PROXY_AUTH || "admin:opencast",
-        }),
-    );
+module.exports = app => {
+    app.use([
+        "/acl-manager",
+        "/admin-ng",
+        "/api",
+        "/info",
+        "/services",
+        "/staticfiles",
+        "/sysinfo",
+        "/ui",
+    ], middleware);
 };
+
+const middleware = process.env.PROXY || process.env.PROXY_TARGET || process.env.PROXY_AUTH
+    ? createProxyMiddleware({
+        target: process.env.PROXY_TARGET || "http://localhost:8080",
+        changeOrigin: true,
+        auth: process.env.PROXY_AUTH || "admin:opencast",
+    })
+    : (req, res, next) => {
+        if (req.method === "POST") {
+            res.status(201);
+        }
+
+        req.url = `/${req.method}/${req.originalUrl}`;
+
+        if (req.method !== "GET") {
+            req.originalMethod = req.method;
+            req.method = "GET";
+            setTimeout(testFiles, 1000, req, res, next);
+        } else {
+            testFiles(req, res, next);
+        }
+    };
+
+const testFiles = express.static(path.join(__dirname, "../../test"));
```

Comment:
The path is missing the `/app`:
```suggestion
const testFiles = express.static(path.join(__dirname, "../../test/app"));
```

Without this change, running `npm start` will immediately end in an error.

Reply:
And this was probably missed when separating this from #399.